src/models/User.ts

```diff
@@ -1,88 +1,48 @@
-import bcrypt from 'bcrypt-nodejs';
-import crypto from 'crypto';
-import mongoose from 'mongoose';
+import mongoose, {Schema} from 'mongoose';
 
 export type UserDocument = mongoose.Document & {
-    email: string;
-    password: string;
-    passwordResetToken: string;
-    passwordResetExpires: Date;
-
-    facebook: string;
-    tokens: AuthToken[];
-
-    profile: {
+    name: string;
+    email: string; // later down the line, we will make this optional for recipients
+    pushTokens: string[]; // expo push tokens
+    donorInfo?: {
+        // all of these attributes are about the business
+        // these attributes are required if donorInfo is present
         name: string;
-        gender: string;
-        location: string;
-        website: string;
-        picture: string;
+        phone: string;
+        address: string;
+        longitude: number;
+        latitude: number;
     };
-
-    comparePassword: comparePasswordFunction;
-    gravatar: (size: number) => string;
+    volunteerInfo?: {
+        phone: string;
+    };
+    recipient: boolean;
+    admin: boolean; // in reality, admin access will be based on the Auth0 token, not this attribute
 };
 
-type comparePasswordFunction = (candidatePassword: string, cb: (err: any, isMatch: any) => {}) => void;
-
-export interface AuthToken {
-    accessToken: string;
-    kind: string;
-}
-
 const userSchema = new mongoose.Schema<UserDocument>({
-    email: { type: String, unique: true },
-    password: String,
-    passwordResetToken: String,
-    passwordResetExpires: Date,
-
-    facebook: String,
-    twitter: String,
-    google: String,
-    tokens: Array,
-
-    profile: {
-        name: String,
-        gender: String,
-        location: String,
-        website: String,
-        picture: String
-    }
+    name: String,
+    email: { type: String, unique: true }, // later down the line, we will make this optional for recipients
+    pushTokens: [String], // expo push tokens
+    donorInfo: { type: new Schema({
+            // all of these attributes are about the business
+            // these attributes are required if donorInfo is present
+            name: String,
```

Comment:
should you add `required: true` to these?

Reply:
Good idea!